sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -1178,19 +1179,19 @@ case class ScalaUDF(
 
   private[this] val resultConverter = catalystConverter
 
-  lazy val udfErrorMessage = {
-    val funcCls = Utils.getSimpleName(function.getClass)
-    val inputTypes = children.map(_.dataType.catalogString).mkString(", ")
-    val outputType = dataType.catalogString
-    s"Failed to execute user defined function($funcCls: ($inputTypes) => $outputType)"
+  lazy val (funcCls, inputTypesString, outputType) = {
+    (Utils.getSimpleName(function.getClass),
+      children.map(_.dataType.catalogString).mkString(", "),
+      dataType.catalogString)
   }
 
   override def eval(input: InternalRow): Any = {
     val result = try {
       f(input)
     } catch {
       case e: Exception =>
-        throw new SparkException(udfErrorMessage, e)
+        throw QueryExecutionErrors.failedExecuteUserDefinedFunctionError(
+          funcCls, inputTypesString, outputType, e)
```

Comment:
We don't need to change the `udfErrorMessage` here. We can just directly pass the message itself.

Reply:
Because we want group exception messages in `QueryExecutionErrors`. We should write msg into `QueryExecutionErrors`.